### ISSUE TITLE
Call to an undefined method, that is defined

### ISSUE THREAD

**Issue by JoeCianflone** [nunomaduro/larastan#571]
- Larastan Version: 0.5.8
- `--level` used: 5 -- but I can replicate this at 2

### Description
I'll be honest, I'm not sure if I'm doing something wrong in my configuration or not, but this feels like an issue for Larastan. I'm using the repository pattern so I've got an `EloquentRepository` that gets extended.

Here's the `EloquentRepository`

```php
<?php

declare(strict_types=1);

namespace App\Domain\Repositories;

use App\Helpers\QueryCache;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Collection;
use Ramsey\Uuid\Uuid;

class EloquentRepository
{
    protected Model $model;
    protected QueryCache $cache;
    protected string $cacheKey = '';

    protected array $withData = [];

    public function __construct()
    {
        $this->cache = resolve('QueryCache');
        $this->cache->setKey($this->cacheKey);
    }
```

In my extending class:

```php
<?php

declare(strict_types=1);

namespace App\Domain\Repositories;

use App\Domain\Models\Listing;

class EloquentListingRepository extends EloquentRepository
{
    public function __construct(Listing $model)
    {
        parent::__construct();
        $this->model = $model;
    }

    public function hasSlug(string $slug): bool
    {
        return $this->model->where('slug', '=', $slug)->count() > 0;
    }

```

### Laravel code where the issue was found

```php

    public function hasSlug(string $slug): bool
    {
        return $this->model->where('slug', '=', $slug)->count() > 0;
    }
```

So this code runs fine, but when I run phpstan I get:

```
  21     Call to an undefined method Illuminate\Database\Eloquent\Model::where().  
```

`Listing` is a descendant of `Model` too. This is getting rather verbose so let me know if you need something else from me...or if I'm just totally screwing something up.

**Comment by canvural** [comment]
Hi,

Yes this is a known issue. And I think it'll be fixed with #565

**Comment by szepeviktor** [comment]
Very tricky!

You need

```yaml
parameters:
    inferPrivatePropertyTypeFromConstructor: true
```

**Comment by JoeCianflone** [comment]
@szepeviktor Tried that it didn't make a difference

**Comment by JoeCianflone** [comment]
@canvural Thanks!

**Comment by canvural** [comment]
@JoeCianflone #565 was merged. Can you try this again on `master`?

**Comment by JoeCianflone** [comment]
@canvural I'll check this out tonight for sure and I'll let you know. 

Thanks!

**Comment by JoeCianflone** [comment]
@canvural This seems to be fixed with #565 I've been kinda pulling apart my codebase for the past couple of days so I've got some other issues I need to resolve, but it's looking good!
